Linker error when building google_iot_mqtt sample with zephyr-sdk 0.11.1

**Describe the bug**
After upgrading to zephyr-sdk 0.11.1 I got a linker error when building `samples/net/cloud/google_iot_mqtt`. The linker complains about `_gettimeofday` and seems to be the same error as in https://github.com/zephyrproject-rtos/zephyr/issues/13384. 

**To Reproduce**
Steps to reproduce the behavior:
1. west build -b nrf52840_pca10056 -d build/iot-sample zephyr/samples/net/cloud/google_iot_mqtt

**Screenshots or console output**
```
/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: /opt/zephyr-sdk/arm-zephyr-eabi/arm-zephyr-eabi/lib/thumb/v7e-m/nofp/libc_nano.a(lib_a-gettimeofdayr.o): in function `_gettimeofday_r':
gettimeofdayr.c:(.text._gettimeofday_r+0xe): undefined reference to `_gettimeofday'
collect2: error: ld returned 1 exit status
```

**Environment (please complete the following information):**
 - OS: ArchLinux
 - Toolchain: zephyr-sdk 0.11.1
 - SHA: f21aaeffaa5ed10629915a9d198ff483865f7be9

--- a/lib/libc/newlib/libc-hooks.c
+++ b/lib/libc/newlib/libc-hooks.c
@@ -358,3 +358,13 @@ void *_sbrk_r(struct _reent *r, int count)
 	return _sbrk(count);
 }
 #endif /* CONFIG_XTENSA */
+
+struct timeval;
+
+int _gettimeofday(struct timeval *__tp, void *__tzp)
+{
+	ARG_UNUSED(__tp);
+	ARG_UNUSED(__tzp);
+
+	return -1;
+}

